### PR TITLE
[16.0][IMP] auditlog: prevent creating logs without log lines

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
This PR enhances the change detection logic in the write_full and write_fast methods to ensure logs are only generated when significant data changes occur. Previously, compute functions could trigger logging even without actual data modifications, leading to logs that lacked corresponding log lines. This update prevents such redundant logging, optimizing performance and maintaining cleaner log records.

@qrtl QT4483

### CODE REVIEW

**Review by kanda999 — APPROVED** [review]
Functional review: LGTM

**Comment by AungKoKoLin1997** [comment]
@oca-server-tools-maintainers
Can you please take a look this PR?

File: auditlog/models/rule.py [hunk 1]
```diff
@@ -400,7 +400,13 @@ def write_full(self, vals, **kwargs):
...
                 .read(fields_list)
             }
-            if self.env.user in users_to_exclude:
+            # Check if any value has actually changed
+            changes = any(
+                old_values[rec_id] != new_values[rec_id]
+                for rec_id in self.ids
+                if rec_id in old_values and rec_id in new_values
+            )
+            if not changes or self.env.user in users_to_exclude:
```
Comment (yostashiro) [inline auditlog/models/rule.py:409]:
```suggestion
            if new_values == old_values or self.env.user in users_to_exclude:
```

File: auditlog/models/rule.py [hunk 2]
```diff
@@ -423,8 +429,14 @@ def write_fast(self, vals, **kwargs):
...
             new_values = {id_: vals2 for id_ in self.ids}
+            # Check if any value has actually changed
+            changes = any(
+                old_values[rec_id] != new_values[rec_id]
+                for rec_id in self.ids
+                if rec_id in old_values and rec_id in new_values
+            )
             result = write_fast.origin(self, vals, **kwargs)
-            if self.env.user in users_to_exclude:
+            if not changes or self.env.user in users_to_exclude:
```
Comment (yostashiro) [inline auditlog/models/rule.py:439]: Please check if this doesn't skip log creation when a value is changed to false from something else.

**Review by yostashiro — CHANGES_REQUESTED** [review]

**Comment by AungKoKoLin1997** [comment]
This improvement is not valid. I misunderstood compute function behavior and confused. Closing this PR.

**Pull request closed without merging**